### ISSUE TITLE
Support custom units for custom measurements

### ISSUE THREAD

**Issue by marandaneto** [getsentry/sentry-dart#1053]
### Description

Due to the limitations of `enums` on Dart, there was no easy way to extend the enum and accept a plain String.
We can leverage this feature if needed, when we upgrade to Dart 2.17, or just set the `unit` as a plain string and document all the possible values.

Related to https://github.com/getsentry/sentry-dart/pull/1011

Example on Java:

```
transaction.setMeasurement("metric1", 2, MeasurementUnit.Duration.DAY)
transaction.setMeasurement("metric1", 2, MeasurementUnit.Custom("myOwnUnit"))
```

**Comment by marandaneto** [comment]
On v7, we can do this:

```
environment:
  sdk: '>=2.17.0 <3.0.0'
  flutter: '>=3.0.0'
```

So we can use new features and unblock this, [Flutter people upgrade fast](https://twitter.com/timsneath/status/1551455025435590656).

**Comment by marandaneto** [comment]
Example: https://github.com/getsentry/sentry-java/blob/e2433098d0addc8570750857f9f20c33ab28143c/sentry/src/main/java/io/sentry/MeasurementUnit.java

**Comment by denrase** [comment]
@marandaneto  I'm not sure how the new enum features of the Language are of any help here. While enum cases can take associated values, they have to be declared with the enum case, and cannot be instantiated freely with any value. So this is different than enums in Swift or sealed classes in Kotlin.  Am I missing something here?

https://dart.dev/guides/language/language-tour#declaring-enhanced-enums

**Comment by marandaneto** [comment]
> @marandaneto I'm not sure how the new enum features of the Language are of any help here. While enum cases can take associated values, they have to be declared with the enum case, and cannot be instantiated freely with any value. So this is different than enums in Swift or sealed classes in Kotlin. Am I missing something here?
> 
> [dart.dev/guides/language/language-tour#declaring-enhanced-enums](https://dart.dev/guides/language/language-tour#declaring-enhanced-enums)

I have not tried using the new enum spec on Dart, but I know its more powerful than the previous version, so isn't possible to do what Java/.NET did?

**Comment by denrase** [comment]
@marandaneto I split it up into multiple enums and a class for the custom measurements, like on Java.

**Comment by marandaneto** [comment]
> @marandaneto I split it up into multiple enums and a class for the custom measurements, like on Java.

That's fine, in the previous version, an `enum` could not `implements` an interface/abstract class, and now it is.
